vault/core.go

```diff
@@ -257,6 +257,12 @@ type Core struct {
 	// token store is used to manage authentication tokens
 	tokenStore *TokenStore
 
+	// identityStore is used to manage client entities
+	identityStore *identityStore
+
+	// entitiesLock is used to ensure thread safety for restoring entities
+	entitiesLock sync.Mutex
```

Comment:
I believe this is unnecessary, because AFAICT loading entities only happens while the Core's `stateLock` is held with a write lock.